jupyter_server/services/contents/manager.py

```diff
@@ -184,6 +298,11 @@ def _default_checkpoints_kwargs(self):
         """,
     )
 
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._pre_save_hooks = []
+        self._post_save_hooks = []
+
```

Comment:
Apologies to nitpick here... 

I think the traitlets way to declare these items is something like:

```python
_pre_save_hooks = List()
_post_save_hooks = List()
```

instead of calling `super` on `__init__`

Reply:
But we don't want them to be traits, do we? The API consists of the `ContentsManager` methods:
- `register_pre_save_hook(hook)`
- `register_post_save_hook(hook)`

Which will append to the private lists.
Or am I missing something?

Reply:
We don't want them to be _configurable_ traits that can be set by the Jupyter config system, i.e. `.tag(config=True)`. 

Defining them as traits is fine. Traits behave exactly like instance attributes with type validation. This just saves you from having to define `__init__` and call `super().__init__(...)`.

Reply:
I'm fine if you'd rather leave this as is. This is just a stylistic choice. I'm approved the PR as-is, so feel free to merge when you're ready.

Reply:
Thanks, I'll make the change.